### PR TITLE
Solved setWill() cause mqtt could not connect with SSL enable #107

### PR DESCRIPTION
**Pull request by moritzj29**
https://github.com/marvinroger/async-mqtt-client/issues/107

also see discussion here for esphome with MQTT fingerprint enabled: https://github.com/esphome/issues/issues/786

### CODE REVIEW

File: src/AsyncMqttClient.hpp [hunk 1]
```diff
@@ -88,6 +88,7 @@ class AsyncMqttClient {
   uint32_t _lastClientActivity;
   uint32_t _lastServerActivity;
   uint32_t _lastPingRequestTime;
+  char _sendbuffer[100];
```
Comment (OttoWinter) [inline src/AsyncMqttClient.hpp]: Why declare this as a new class member variable if it's only used in one context? And why then declare it as size 100 bytes when only 12 are used?

Please move this declaration into the `_onConnect` call just before it's used and change its size to 12 bytes.

**Comment by moritzj29** [comment]
I'm sorry, I cannot comment on any of the changes made, since I did not make them. This is a pending PR of the base repository by @jeroenst.

Merging these changes into async-mqtt-client for esphome helped solving my issues described earlier. Shame on me, I have not worked through the code of these libraries. I'm just thankfully using them.

**Comment by jeroenst** [comment]
Good points @ottowinter.

I will change it in the following days.

**Comment by jeroenst** [comment]
Ok, I changed the code as commented by @OttoWinter

**Review by OttoWinter — APPROVED** [review]
Thanks!

**Pull request merged**